src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -107,35 +110,66 @@ export default Backbone.Model.extend({
       [dateType]: this.formatDateRange(relativeRange, unit, dateFormat),
     };
   },
+  getDefaultStatesFilter() {
+    const isDoneOnly = this.id === 'done-last-thirty-days';
+    const { done, notDone } = this.states.groupByDone();
+
+    return isDoneOnly ? done : notDone;
+  },
+  setDefaultFilterStates() {
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+    this.set({ filters: { states: defaultStatesFilter.map('id') } });
+  },
+  getSelectedStates() {
+    const defaultStatesFilterIds = this.getDefaultStatesFilter().map('id');
+    const selectedStates = this.getFilters().states;
+
+    // remove any invalid state ids (i.e. ids that don't belong to any default states)
+    return filter(selectedStates, id => contains(defaultStatesFilterIds, id)).join();
+  },
+  getFiltersCount() {
+    const filtersState = this.get('filters');
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+
+    const customFiltersState = omit(filtersState, 'states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterSize = size(filtersState.states);
+    const defaultStatesSize = size(defaultStatesFilter.map('id'));
+
+    const statesFilterCount = statesFilterSize !== defaultStatesSize ? 1 : 0;
+
+    return customFiltersSize + statesFilterCount;
+  },
```

Comment:
Added this function to the state, but not using it yet.

`filters_app.js` and `filters-sidebar_app.js` don't have access to any of the state methods. Since their state is only `this.state.filters` (not the entire `this.state`).

So in those files, `this.getState().getFiltersCount()` doesn't work and results in an error... 🤔

Reply:
I mentioned this in my review prior to reading this, but yeah we just need to make a defined state model for those other apps with this function